### PR TITLE
Fix Smith CI

### PR DESCRIPTION
**Pull request by chapman39**
Smith CI has been running only Serac tests due to an issue with using symlinks! This MR attempts to resolve this.

EDIT: Turns out the fix was within Smith. The remaining changes here are minor cleanups.

### CODE REVIEW

**Review by tupek2 — APPROVED** [review]

**Review by white238 — APPROVED** [review]

**Comment by white238** [comment]
This being broken in this way actually has more implications than we think. Anywhere that `get_repo_dir` is called needs to be reviewed or that function needs to be fixed.

**Comment by white238** [comment]
Good catch btw....

**Comment by chapman39** [comment]
The previous solution didn't solve the problem all the way, since there were issues with `get_project_name()` returning serac instead of smith.

This new solution modified `get_repo_dir()` to find the smith dir if it's a serac submodule. The only catch with this solution is that you won't be able to build and test only serac if it's a submodule of smith.

File: scripts/llnl/common_build_functions.py [hunk 1]
```diff
@@ -637,9 +637,26 @@ def get_build_dir(prefix, host_config):
...
+    global _repo_dir
+    if not _repo_dir:
+        # Set repo dir to be relative to this script
+        _repo_dir = os.path.abspath(pjoin(get_script_dir(), "../.."))
+
+        # Check the parent dir for an uberenv config with a package_name "smith".
+        # This means Serac is a submodule. Change the repo dir to be parent dir (i.e.
+        # Smith repo dir)
+        smith_uberenv_config_path = pjoin(_repo_dir, "../.uberenv_config.json")
+        if os.path.exists(smith_uberenv_config_path):
```
Comment (white238) [inline scripts/llnl/common_build_functions.py:653]: Should use `get_project_name()` here...

File: scripts/llnl/common_build_functions.py [hunk 2]
```diff
@@ -637,9 +637,26 @@ def get_build_dir(prefix, host_config):
...
+            with open(smith_uberenv_config_path) as json_file:
```
Comment (white238) [inline scripts/llnl/common_build_functions.py]: Should use get_project_name() here...

Comment (chapman39) [reply scripts/llnl/common_build_functions.py]: The problem with that is `get_project_name()` depends on `get_repo_dir()`. I could move the `.uberenv_config.json` query to a separate function though.

**Review by white238 — APPROVED** [review]

**Pull request merged**